src/Component/ComponentFunctions.jl

```diff
@@ -0,0 +1,39 @@
+# Declare Common Functions so that they can be dispatched from ModuleExtensions
+function add_collision_event end
```

Comment:
You could add docstrings to this file to describe the semantic meaning of the function in general. Then add additional docstrings for more specific documentation about the method implemented for each type of component. 

But I don't know enough about how you're using these to document that for you atm